runner/execer/os/os_execer.go

```diff
@@ -51,27 +51,49 @@ func (e *osExecer) Exec(command execer.Command) (result execer.Process, err erro
 	}
 
 	cmd := exec.Command(command.Argv[0], command.Argv[1:]...)
+	cmd.Dir = command.Dir
+
+	// Sets pgid of all child processes to cmd's pid
+	cmd.SysProcAttr = &syscall.SysProcAttr{Setpgid: true}
 
-	cmd.Stdout, cmd.Stderr, cmd.Dir = command.Stdout, command.Stderr, command.Dir
 	// Make sure to get the best possible Writer, so if possible os/exec can connect
 	// the command's stdout/stderr directly to a file, instead of having to go through
 	// our delegation
-	if stdoutW, ok := cmd.Stdout.(WriterDelegater); ok {
-		cmd.Stdout = stdoutW.WriterDelegate()
+	if stdoutW, ok := command.Stdout.(WriterDelegater); ok {
+		command.Stdout = stdoutW.WriterDelegate()
 	}
 	if stderrW, ok := cmd.Stderr.(WriterDelegater); ok {
-		cmd.Stderr = stderrW.WriterDelegate()
+		command.Stderr = stderrW.WriterDelegate()
 	}
 
-	// Sets pgid of all child processes to cmd's pid
-	cmd.SysProcAttr = &syscall.SysProcAttr{Setpgid: true}
+	// Use pipes due to possible hang in process.Wait().
+	// See: https://github.com/noxiouz/stout/commit/42cc533a0bece540f2424faff2a960876b21ffd2
+	stdErrPipe, err := cmd.StderrPipe()
+	if err != nil {
+		return nil, err
+	}
+	stdOutPipe, err := cmd.StdoutPipe()
+	if err != nil {
+		return nil, err
+	}
+	var wg sync.WaitGroup
+	wg.Add(2)
```

Comment:
super duper nit: might be more clear to wg.Add(1) in each goroutine

Reply:
I believe it needs to be done outside of the goroutines.